tests/Unit/Pypp/Pypp.hpp

```diff
@@ -71,16 +72,51 @@ template <typename T>
 using convert_to_container_of_doubles_t =
     typename convert_to_container_of_doubles<T>::type;
 
+template <typename T>
+struct SliceContainerImpl {
+  static auto apply(const T& container_dv, const size_t slice_idx) noexcept {
+    convert_to_container_of_doubles_t<std::decay_t<decltype(container_dv)>>
+        container_double{};
+    ASSERT(slice_idx < container_dv.begin()->size(),
+           "Trying to slice DataVector of size " << container_dv.begin()->size()
+                                                 << "with slice_idx "
+                                                 << slice_idx);
+    for (decltype(auto) double_and_datavector_components :
+         boost::combine(container_double, container_dv)) {
+      boost::get<0>(double_and_datavector_components) =
+          boost::get<1>(double_and_datavector_components)[slice_idx];
+    }
+    return container_double;
+  }
+};
+
+template <>
+struct SliceContainerImpl<double> {
+  static double apply(const double& t, const size_t /*slice_index*/) noexcept {
+    return t;
+  }
+};
+
+template <typename T>
+decltype(auto) slice_container_of_datavectors_to_container_of_doubles(
+    const T& in, const size_t slice_index) noexcept {
+  return SliceContainerImpl<T>::apply(in, slice_index);
+}
+
 template <typename R>
 struct CallImpl<
     R, Requires<(tt::is_a_v<Tensor, R> or tt::is_std_array_v<R>) and
                  cpp17::is_same_v<typename R::value_type, DataVector>>> {
   template <typename... Args>
   static R call(const std::string& module_name,
                 const std::string& function_name, const Args&... t) {
-    static_assert(sizeof...(Args) > 0,
-                  "Call to python which returns a Tensor of DataVectors must "
-                  "pass at least one Tensor of DataVectors.");
+    static_assert(
+        sizeof...(Args) > 0 and
+            cpp17::is_same_v<typename decltype(
+                                 get_first_argument(t...))::value_type,
```

Comment:
This static_assert only half works.  If the sizeof...(Args) test is false the second test fails to compile so the assertion is not triggered.

Reply:
Ah I see. Should I use two static asserts then to ensure the first condition gets triggered?

Reply:
Yes, two static_asserts will do what you want. Put the size one first so the compiler hits it first :)